docs/docs/tutorial/creating-a-source-plugin/part-4/index.mdx

```diff
@@ -0,0 +1,865 @@
+
+    ```shell
+    verbose [plugin] Last fetched date: 1677669643651
+    ```
+
+    Great, it's working!
+
+    <Announcement>
+
+    **Pro tip:** If you'd save files with the `cache` API you could also inspect what is saved by going to the `.cache/caches` folder and looking for a folder named like your plugin. For this tutorial the path would be `site/.cache/caches/plugin`.
+
+    </Announcement>
+
+## `pluginOptionsSchema` API
+
+In most cases a Gatsby source plugin needs one or more options as it interacts with third-party APIs that require customization and authentication (e.g. different endpoints or API keys). You don't want other users to use your credentials, so you'll need a way for your users to provide theirs. That's where plugin options come in!
+
+The [Configuring Plugin Usage with Plugin Options](/docs/how-to/plugins-and-themes/configuring-usage-with-plugin-options/) guide goes into depth about plugin options and the [`pluginOptionsSchema` API](/docs/reference/config-files/gatsby-node/#pluginOptionsSchema) so if you want to know more beyond this guide, be sure to give it a read. But to give a short summary:
+
+- A Gatsby plugin with options included makes those options available in the second argument of Gatsby [Node](/docs/reference/config-files/gatsby-node/), [Browser](/docs/reference/config-files/gatsby-browser/), and [SSR](/docs/reference/config-files/gatsby-ssr/) APIs.
+
+    A sample `gatsby-config` file:
+
+    ```js:title=gatsby-config.js
+    module.exports = {
+      plugins: [
+        {
+          resolve: `gatsby-plugin-console-log`,
+          options: { message: "Hello world" },
+        },
+      ],
+    }
+    ```
+
+    Usage of that option in a Node API:
+
+    ```js:title=plugins/gatsby-plugin-console-log/gatsby-node.js
+    exports.onPreInit = (_, pluginOptions) => {
+      console.log(
+        `logging: "${pluginOptions.message}" to the console` // highlight-line
+      )
+    }
+    ```
+- A plugin can optionally define a schema (through the `pluginOptionsSchema` API) to enforce a type for each option. Gatsby will validate that the options users pass match the schema to help them correctly set up their site.
+
+    ```js:title=plugins/gatsby-plugin-console-log/gatsby-node.js
+    exports.pluginOptionsSchema = ({ Joi }) => {
+      return Joi.object({
+        message: Joi.string()
+          .required()
+          .description(`The message logged to the console.`),
+      })
+    }
+    ```
+
+Here are some tips when working with plugin options:
+
+- **Consider the options a public API contract between your users and your plugin.** Follow [SemVer](https://semver.org/) versioning when adding, modyfing, and removing options. For example, if you want to remove an option you should only do this in a major release of your plugin since it's a breaking change.
```

Comment:
```suggestion
- **Consider the options a public API contract between your users and your plugin.** Follow [SemVer](https://semver.org/) versioning when adding, modifying, and removing options. For example, if you want to remove an option you should only do this in a major release of your plugin since it's a breaking change.
```